Aviation extension

espinielli opened the pull request:
I propose this extension with aviation data.

The best approach would probably be to provide a mechanism for extensions, similar to #55 but not limited to regions only...

vincentarelbundock commented on R/countrycode_data.R:
```diff
@@ -12,6 +12,7 @@
 #'   \item imf: International Monetary Fund
 #'   \item iso2c: ISO-2 character
 #'   \item iso3c: ISO-3 character
+#'   \item iso3c: ISO-3 character
```
duplicate entry?

vincentarelbundock commented on data/aviation_data.csv:
```diff
@@ -0,0 +1,274 @@
+Austria,LO,L,ESRA North-West,Eurocontrol
+Austria-Hungary,"NA","NA","NA","NA"
+Azerbaijan,UB,U,Other Europe,Asia
+Baden,"NA","NA","NA","NA"
```
why are NAs in quotes?

espinielli replied:
I exported with write_csv from readr...probably a default there.

vincentarelbundock commented on data/aviation_data.csv:
```diff
@@ -0,0 +1,274 @@
+country_name,icao,icaoregion,statfor,pru
```
What is the rationale for not including those directly to the original countrycode_data.csv?

espinielli replied:
I see there are other PR, i.e. the one with official UN country short names, and I did want to make the process to add specific extensions easier in case this will not get into the official release.
Please let me know whether you want it merged.

vincentarelbundock commented:
I'm happy to merge it. However, I would prefer a minimalist change. Just modify the CSV file to add a new column (with proper non-quoted NAs), and add adjust the documentation accordingly. No unnecessary additional files. 

Also, in terms of adding new codes, how about just merging them to countrycode_data:

```
library(dplyr)
library(countrycode)
countrycode::countrycode_data = left_join(countrycode::countrycode_data, aviation_data)

countrycode('AG', 'aviation_code', 'country.name')
```

Did that not work for you?

espinielli commented:
I did a full_join in `prep.R` because I had more observations than original countrycode_data.
I'see for a minimalistic PR.
Thanks

espinielli commented:
I think I did what requested: a simpler set of modifications.
* added aviation columns to `countrycode_data.csv`
* corrected typo in `man`
* removed separate `data/aviation_data.csv`
* reverted `prep.R`

espinielli commented:
At this point it looks ok to me.
Please let me know whether you think something else is needed.

vincentarelbundock commented:
why did you reformat all the code (change indentation, etc.)? This makes it much harder to see what's changed and what hasn't...

vincentarelbundock commented:
Working on a solution now

espinielli commented:
@vincentarelbundock I think this can be closed (once 1.19 is delivered) given that the provision for a custom dictionary as in #67 and the example as in the README, make the aviation mappings possible.
(Of course It would be nice if the `aviation.csv` file would be part of package installation...
and used in the examples of the API/documentation)

vincentarelbundock commented:
Thanks! Life is crazy right now, but will try to put together a release as soon as possible.

espinielli commented:
No problem, take your time...in the meantime I have installed from github and used in my other project (mapping [flight flows in/out of and internal to Europe](https://github.com/espinielli/flowsr)).
Thanks.
